src/ContainerCommitModal.jsx

```diff
@@ -10,54 +10,37 @@ import * as utils from './util.js';
 import * as client from './client.js';
 import { ErrorNotification } from './Notification.jsx';
 import { fmt_to_fragments } from 'utils.jsx';
-import { DialogsContext } from "dialogs.jsx";
+import { useDialogs } from "dialogs.jsx";
 
 const _ = cockpit.gettext;
 
-class ContainerCommitModal extends React.Component {
-    static contextType = DialogsContext;
-
-    constructor(props) {
-        super(props);
-        this.state = {
-            imageName: "",
-            tag: "",
-            author: "",
-            command: props.container.Command ? utils.quote_cmdline(props.container.Command) : "",
-            pause: false,
-            selectedFormat: "oci",
-            commitInProgress: false,
-            useDocker: false,
-            nameError: "",
-        };
-
-        this.handleInputChange = this.handleInputChange.bind(this);
-        this.handleCommit = this.handleCommit.bind(this);
-    }
-
-    handleInputChange(targetName, value) {
-        const newState = { [targetName]: value };
-
-        if (targetName === "imageName" || targetName === "tag")
-            newState.nameError = "";
-
-        this.setState(newState);
-    }
-
-    handleCommit(force) {
-        const Dialogs = this.context;
-
-        if (!force && !this.state.imageName) {
-            this.setState({ nameError: "Image name is required" });
+const ContainerCommitModal = ({ container, localImages }) => {
+    const Dialogs = useDialogs();
+
+    const [imageName, setImageName] = useState("");
+    const [tag, setTag] = useState("");
+    const [author, setAuthor] = useState("");
+    const [command, setCommand] = useState(container.Command ? utils.quote_cmdline(container.Command) : "");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1146-20221124-132403-fa425528-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)